Added nodeSelector to deployments

atropos112 opened the pull request:
I am in need of node selector, because my cluster has nodes with amd64 as well as arm64 architectures and rocketchat doesn't have arm64 images so was crashlooping.

I can't see a single reason not to add this to main line and let others use this when they please, hence the PR. The issue made here: https://github.com/RocketChat/helm-charts/issues/99 

Note the mongodb also has to have nodeSelector set but its already present in that helm chat.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/RocketChat/helm-charts?pullRequest=98) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/RocketChat/helm-charts?pullRequest=98) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/RocketChat/helm-charts?pullRequest=98) it.</sub>

ChrisJBurns commented:
Can confirm we need `nodeSelector` functionality also. I can close https://github.com/RocketChat/helm-charts/issues/97 when this PR is merged

ChrisJBurns commented:
@atropos112  I can't see the `values.yaml` file being updated?

atropos112 commented:
Thanks @ChrisJBurns , I added the nodeSelector bit to values. Was testing by changing value.yaml file and once I felt everything worked i "git restored" the values.yaml forgetting there was a change there, thanks ! Added nodeSelector to values.yaml now.
